### PR TITLE
respect mode text color

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
**Description:**
Makes sure mode based text color overrides are used on the map

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

File: lib/actions/apiV2.js [hunk 1]
```diff
@@ -1144,6 +1145,16 @@ export function routingQuery(searchId = null, updateSearchInReducer) {
                               config.transitOperators
                             ),
                             { color: leg?.route?.color, mode: leg.mode }
+                          ).split('#')?.[1],
+                          textColor: getRouteTextColorBasedOnSettings(
```
Comment (daniel-heppner-ibigroup) [inline lib/actions/apiV2.js:1155]: put this in a const at the top of the map. also refactor `getRouteOperator` into its own variable so it isn't repeated.

File: lib/util/viewer.js [hunk 2]
```diff
@@ -353,6 +353,10 @@ export function getRouteColorBasedOnSettings(operator = {}, route = {}) {
       return backgroundColor
   }
 }
+export function getRouteTextColorBasedOnSettings(operator = {}, route = {}) {
```
Comment (daniel-heppner-ibigroup) [inline lib/util/viewer.js:358]: delete this change and revert the file

**Review by daniel-heppner-ibigroup — CHANGES_REQUESTED** [review]

**Review by daniel-heppner-ibigroup — APPROVED** [review]
alright, this is looking good now. thanks for the fix

**Review by amy-corson-ibigroup — APPROVED** [review]
Beautiful colors!

**Pull request merged**